packages/api-queries/src/site.ts

```diff
@@ -15,8 +15,15 @@ import type { Query } from '@tanstack/react-query';
 
 const KNOWN_ERRORS = [ 'unknown_blog', 'unauthorized' ];
 
-export const siteBySlugQuery = ( siteSlug: string ) =>
-	queryOptions( {
+export const siteBySlugQuery = ( siteSlug: string ) => {
+	// Used to find an existing Site object which is already in the `site-by-id` cache.
+	const getFromCache = () =>
+		queryClient
+			.getQueriesData< Site >( { queryKey: [ 'site-by-id' ] } )
```

Comment:
I think this must be exactly this otherwise it might get stale object whenever we update `SITE_FIELDS` or `SITE_OPTIONS` (not sure)


```suggestion
			.getQueriesData< Site >( siteByIdQuery( site.ID ) )
```

Reply:
You're probably right that I should make sure the fields and options match, but I'll have to do that in the `.find` function. I don't have the `site.ID` in this situation which is why I'm having to query the entire `site-by-id` cache here.

Reply:
Oh... you're right, sorry. Is it feasible? It should be, right? 😄

Reply:
Addressed feedback here f3afde1cbd6ed0184fe6a7f481bb314290742e5a

I'm calling `getQueriesData` with a predicate function now so that I can check all the parts of the query key that I have. I only perform the more expensive `SITE_FIELDS` and `SITE_OPTIONS` checks once I'm sure I've found the correct site.